ssl/ssl_text.cc

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 1995-2022 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 2005 Nokia. All rights reserved.
+ */
+
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+// Modifications Copyright Amazon.com, Inc. or its affiliates.
+
+
+#include <openssl/ssl.h>
+
+#include "internal.h"
+
+
+int SSL_SESSION_print(BIO *bp, const SSL_SESSION *sess) {
+  if (sess == nullptr) {
+    return 0;
+  }
+
+  bool tls13 = (sess->ssl_version == TLS1_3_VERSION);
+  if (BIO_puts(bp, "SSL-Session:\n") <= 0) {
+    return 0;
```

Comment:
Should any of these `return 0` lines also have a `OPENSSL_PUT_ERROR`?

Reply:
I don't think this necessarily needs an error since it's a print function for debugging purposes. The original OpenSSL function doesn't push any errors onto the stack either.
* https://github.com/openssl/openssl/blob/3d254b31344e82b8f10fda8bab196757a377eb63/ssl/ssl_txt.c#L32